packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -179,7 +179,12 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
         }
 
         if (openHandler) {
-            await openHandler.open(uri, editorOptions);
+            const widget = await openHandler.open(uri, editorOptions);
+            if (widget instanceof Widget) {
+                if (!widget.isAttached) {
+                    await this.shell.addWidget(widget, editorOptions?.widgetOptions);
+                }
```

Comment:
@debovema, thanks for your contribution! I believe the code is good, but should be moved elsewhere. See below.

It looks like this is should be the responsibility of the open handler for the widget. For example, here's the default `WidgetOpenHandler`:

https://github.com/eclipse-theia/theia/blob/17b419fdf0d1015f2269ffcb64a095b305a8ec2d/packages/core/src/browser/widget-open-handler.ts#L98-L100

But the same is not done when a new custom editor is created:

https://github.com/eclipse-theia/theia/blob/17b419fdf0d1015f2269ffcb64a095b305a8ec2d/packages/plugin-ext/src/main/browser/custom-editors/custom-editor-opener.tsx#L83-L87

I think that for consistency's sake, that's the place to handle this issue.

Reply:
Alternatively, it looks like this is where the attachment is handled, at the moment: 

https://github.com/eclipse-theia/theia/blob/17b419fdf0d1015f2269ffcb64a095b305a8ec2d/packages/plugin-ext/src/main/browser/custom-editors/custom-editors-main.ts#L234-L240

Reply:
Hi @colin-grant-work 

Thank you for your review.

The issue with your suggestion to move the logic in the Custom Editor Opener (*custom-editor-opener.tsx*) is that we don't have the info provided by VSCode extension being opened in this location.

The other place in ```$createCustomEditorPanel``` method in *custom-editors-main.ts* does receive a viewColumn parameter but AFAICS this parameter is hardcoded to ```EditorPosition.ONE```, because of this:

https://github.com/eclipse-theia/theia/blob/17b419fdf0d1015f2269ffcb64a095b305a8ec2d/packages/plugin-ext/src/main/browser/custom-editors/custom-editors-main.ts#L152

If I understand well, the call to ```await openHandler.open(uri, editorOptions);``` actually delegates to the ```CustomEditorOpener``` class (this class is not injectable with inversify so I assume it is created here : https://github.com/eclipse-theia/theia/blob/17b419fdf0d1015f2269ffcb64a095b305a8ec2d/packages/plugin-ext/src/main/browser/custom-editors/plugin-custom-editor-registry.ts#L84

So it might be the simplest solution to add the behaviour in my original place, don't you think ?

Reply:
> The issue with your suggestion to move the logic in the Custom Editor Opener (custom-editor-opener.tsx) is that we don't have the info provided by VSCode extension being opened in this location.

I'm not sure what you mean by this. As you say

> the call to await openHandler.open(uri, editorOptions); actually delegates to the CustomEditorOpener

the call to `openHandler.open` calls into `CustomEditorOpener.open`, so if we have the information we need where you've added it, then we have that very same information in `custom-editor-opener.tsx`, because we're passing that information to the `CustomEditorOpener`, and we have access to the `ApplicationShell` there, as well, so there's nothing in the code you've added that we couldn't do in the `CustomEditorOpener` class.

Regarding the hardcoded `EditorPosition.ONE` and the comment  there, I don't think I understand what they mean. Placing a widget in a particular column doesn't necessarily have anything to do with whether the widget can be split, but I haven't worked much with custom editors, myself, so there may be some logic there that I'm missing. To my mind, the problem has nothing to do with the splittability of editors but rather with whether Theia correctly handles the `viewColumn` field, and there is code that appears to try to do so:

https://github.com/eclipse-theia/theia/blob/d7cdb85c1793ae5e4af8d05da249a2dcea606da2/packages/plugin-ext/src/main/browser/webviews-main.ts#L108-L114

So it may be worth simply swapping out that hardcoded value with whatever is contextually appropriate to see if that addresses the problem.

Reply:
> the call to `openHandler.open` calls into `CustomEditorOpener.open`, so if we have the information we need where you've added it, then we have that very same information in `custom-editor-opener.tsx`, because we're passing that information to the `CustomEditorOpener`, and we have access to the `ApplicationShell` there, as well, so there's nothing in the code you've added that we couldn't do in the `CustomEditorOpener` class.

I pushed another commit moving the logic in the CustomEditorOpener.

Please notice that the options parameter has been changed from ```OpenerOptions``` to ```WidgetOpenerOptions```.
I hope there are no impact elsewhere.

The general behaviour is fine but there are some errors in the console worth investigating :
![image](https://user-images.githubusercontent.com/8082252/148517669-31d8de89-5444-499d-9d37-f2c16b81a2b7.png)

Reply:
@colin-grant-work : eventually I fully reworked the propagation of WidgetOpenerOptions from CustomEditorOpener to CustomEditorsMainImpl

Reply:
@debovema, 

> @colin-grant-work : eventually I fully reworked the propagation of WidgetOpenerOptions from CustomEditorOpener to CustomEditorsMainImpl

Does the latest code cover functionality apart from what's described in #10583, or do the testing steps remain the same?

Reply:
No it covers the exact same original issue.
But the implementation is more consistent.